src/theme/CodeBlock/index.js

```diff
@@ -0,0 +1,35 @@
+import React, { isValidElement } from "react";
+import useIsBrowser from "@docusaurus/useIsBrowser";
+import ElementContent from "@theme/CodeBlock/Content/Element";
+import StringContent from "@theme/CodeBlock/Content/String";
+/**
+ * Best attempt to make the children a plain string so it is copyable. If there
+ * are react elements, we will not be able to copy the content, and it will
+ * return `children` as-is; otherwise, it concatenates the string children
+ * together.
+ */
+function maybeStringifyChildren(children) {
+  if (React.Children.toArray(children).some((el) => isValidElement(el))) {
+    return children;
+  }
+  // The children is now guaranteed to be one/more plain strings
+  return Array.isArray(children) ? children.join("") : children;
+}
+export default function CodeBlock({ children: rawChildren, ...props }) {
+  // The Prism theme on SSR is always the default theme but the site theme can
+  // be in a different mode. React hydration doesn't update DOM styles that come
+  // from SSR. Hence force a re-render after mounting to apply the current
+  // relevant styles.
+  const isBrowser = useIsBrowser();
+  const children = maybeStringifyChildren(rawChildren);
+  const CodeBlockComp =
+    typeof children === "string" ? StringContent : ElementContent;
+  return (
+    <figure aria-roledescription="Code block" tabIndex={0}>
+      <figcaption className="sr-only">Following is a code block</figcaption>
```

Comment:
This figcaption is unnecessary and lengthy as the screen reader can already read the `aria-roledescription` above. It may be useful, if possible, to include an sr-only announcement of the language (e.g. "bash", "hcl2", etc) in a figcaption.

For reference, NVDA reads this as "Code block, caption: following is a code block".

Question: does Voiceover announce the `aria-roledescription`? If not, adding a [role parameter](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) might help.

Reply:
The reason i asked you to check was because on mac, aria-roledescription was not being read by Voice Over. I needed to include the figcaption for that.